pugl/pugl_osx.m

```diff
@@ -439,7 +439,8 @@ - (void) keyDown:(NSEvent*)event
 	const NSPoint      rloc  = [NSEvent mouseLocation];
 	const NSString*    chars = [event characters];
 	const char*        str   = [chars UTF8String];
-	const uint32_t     code  = puglDecodeUTF8((const uint8_t*)str);
+        const PuglKey      special = keySymToSpecial(puglview, event);
+        const uint32_t     code  = special ? 0 : puglDecodeUTF8((const uint8_t*)str);
```

Comment:
Interesting.  What ended up coming out here, out of curiosity?

Reply:
This was created while I implemented a LineEdit widget. The LineEdit widget first checks if a character is to be entered by checking (code != 0). If yes, the character is added to the text. If no character was entered it also checks for arrow keys.

What was happening on macOS is that if you use arrow keys, then the code of the keyboard event was set to the value for the arrow keys, and code was != 0. Instead of having arrow keys move the cursor, the character was added to the text inside the line edit. So this was the reason for setting code for all special keys to zero.

An alternative implementation would be to check if code is in the reserved unicode section (U+E000–U+F8FF) which seems to be where all macOS special key codes are, and unset code for these events.

Reply:
Seems like special and code setting needs to be factored into the same function so this can all be handled correctly in one place?

Reply:
Right. I'd still have to try implementing it to see if its more elegant, but it could be. So whats the strategy now during the merge? Should I commit on top of this branch, or better wait until you merged stuff so far?

Reply:
As you like.  I can never figure out this PR workflow, ideally we'd just work on the same branch...

I can rebase anything, though, so whatever.  I'll try to pop some stuff off and get them on master.  I did some testing on MacOS and it seems fine, I'll try Windows now.